raster/src/main/scala/geotrellis/raster/merge/SinglebandTileMergeMethods.scala

```diff
@@ -154,4 +154,69 @@ trait SinglebandTileMergeMethods extends TileMergeMethods[Tile] {
+    val targetRE = RasterExtent(combinedExtent, targetCS)
+    val mutableTile = ArrayTile.empty(self.cellType, targetRE.cols, targetRE.rows)
+
+    self.cellType match {
+      case BitCellType | ByteCellType | UByteCellType | ShortCellType | UShortCellType | IntCellType  =>
+        val interpolateLeft: (Double, Double) => Int = Resample(method, self, extent, targetCS).resample _
+        val interpolateRight: (Double, Double) => Int = Resample(method, other, otherExtent, targetCS).resample _
+        // Assume 0 as the transparent value
+        cfor(0)(_ < targetRE.rows, _ + 1) { row =>
+          cfor(0)(_ < targetRE.cols, _ + 1) { col =>
+            val (x,y) = targetRE.gridToMap(col, row)
+            val (l,r) = (interpolateLeft(x, y), interpolateRight(x, y))
+            mutableTile.set(col, row, unionInt(Some(l.toDouble), Some(r.toDouble)))
+          }
+        }
+      case FloatCellType | DoubleCellType =>
+        val interpolateLeft: (Double, Double) => Double = Resample(method, self, extent, targetCS).resampleDouble _
+        val interpolateRight: (Double, Double) => Double = Resample(method, other, otherExtent, targetCS).resampleDouble _
+
+        // Assume 0.0 as the transparent value
+        cfor(0)(_ < targetRE.rows, _ + 1) { row =>
+          cfor(0)(_ < targetRE.cols, _ + 1) { col =>
+            val (x,y) = targetRE.gridToMap(col, row)
+            val (l,r) = (interpolateLeft(x, y), interpolateRight(x, y))
+            mutableTile.setDouble(col, row, unionF(Some(l), Some(r)))
+          }
+        }
+      case x if x.isFloatingPoint =>
+        val interpolateLeft: (Double, Double) => Double = Resample(method, self, extent, targetCS).resampleDouble _
+        val interpolateRight: (Double, Double) => Double = Resample(method, other, otherExtent, targetCS).resampleDouble _
+        cfor(0)(_ < targetRE.rows, _ + 1) { row =>
+          cfor(0)(_ < targetRE.cols, _ + 1) { col =>
+            val (x,y) = targetRE.gridToMap(col, row)
+            val l = interpolateLeft(x, y)
+            val r = interpolateRight(x, y)
+            val maybeL = if (isNoData(l)) None else Some(l)
+            val maybeR = if (isNoData(r)) None else Some(r)
+            mutableTile.setDouble(col, row, unionF(maybeL, maybeR))
+          }
+        }
+      case _ =>
+        val interpolateLeft: (Double, Double) => Int = Resample(method, self, extent, targetCS).resample _
+        val interpolateRight: (Double, Double) => Int = Resample(method, other, otherExtent, targetCS).resample _
+        cfor(0)(_ < targetRE.rows, _ + 1) { row =>
+          cfor(0)(_ < targetRE.cols, _ + 1) { col =>
+            val (x,y) = targetRE.gridToMap(col, row)
+            val l = interpolateLeft(x, y)
+            val r = interpolateRight(x, y)
+            val maybeL = if (isNoData(l)) None else Some(l.toDouble)
+            val maybeR = if (isNoData(r)) None else Some(r.toDouble)
+            mutableTile.set(col, row, unionInt(maybeL, maybeR))
```

Comment:
Should not there be extra `isNoData(self.getDouble(col, row))` checks?  (in case it's not implied that it's incapuslated into the union function definition)